airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/cdc/MongoDbCdcInitializer.java

```diff
@@ -183,9 +213,60 @@ public List<AutoCloseableIterator<AirbyteMessage>> createCdcIterators(
             AirbyteStreamStatusTraceMessage.AirbyteStreamStatus.COMPLETE)))
         .toList();
 
-    return Stream.of(Collections.singletonList(initialSnapshotIterator), cdcStreamsStartStatusEmitters,
-        Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
-        cdcStreamsCompleteStatusEmitters).flatMap(Collection::stream).toList();
+    if (startedCdcStreamList.isEmpty()) {
+      LOGGER.info("First sync - no cdc streams have been completed or started");
+      /*
+       * This is the first run case - no initial loads have been started. In this case, we want to run the
+       * iterators in the following order: 1. Run the initial load iterators. This step will timeout and
+       * throw a transient error if run for too long (> 8hrs by default). 2. Run the debezium iterators
+       * with ALL of the incremental streams configured. This is because if step 1 completes, the initial
+       * load can be considered finished.
+       */
+      return Collections.singletonList(
+          AutoCloseableIterators.concatWithEagerClose(
+              Stream.of(
+                  cdcStreamsStartStatusEmitters,
+                  Collections.singletonList(initialSnapshotIterator),
+                  Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
+                  cdcStreamsCompleteStatusEmitters).flatMap(Collection::stream).collect(Collectors.toList()),
+              AirbyteTraceMessageUtility::emitStreamStatusTrace));
```

Comment:
Here and below can we confirm that we are not emitting stream status twice? I think `AirbyteTraceMessageUtility::emitStreamStatusTrace` might also emit stream statuses

Reply:
thanks for verifying this for me.